app/code/VConnect/OrderVolume/Model/ResourceModel/Order/Volume/SaveOrderTotalVolume.php

```diff
@@ -0,0 +1,52 @@
+<?php
+declare(strict_types=1);
+
+namespace VConnect\OrderVolume\Model\ResourceModel\Order\Volume;
+
+use Magento\Sales\Api\Data\OrderInterface;
+use VConnect\OrderVolume\Model\Order\OrderVolumeCalculator;
+use Magento\Framework\App\ResourceConnection;
+use VConnect\OrderVolume\Model\Order\Config as OrderConfig;
+use VConnect\OrderVolume\Model\Order\Item\OrderItemsVolumeCalculator;
+
+class SaveOrderTotalVolume
+{
+    public function __construct(
+        private ResourceConnection $resourceConnection,
+        private OrderVolumeCalculator $orderVolumeCalculator
+    ){}
+
+    /**
+     * @param OrderInterface $order
+     * @param OrderItemsVolumeCalculator $orderItemsVolumeCalculator
+     */
+    public function execute(OrderInterface $order, OrderItemsVolumeCalculator $orderItemsVolumeCalculator): void
+    {
+        $orderID = (int)$order->getEntityId();
+        $orderTotalVolume = $this->getOrderTotalVolume($orderItemsVolumeCalculator);
+        $connection = $this->resourceConnection->getConnection();
+
+        $connection->update(
+            OrderConfig::SALES_ORDER_TABLE,
+            [OrderConfig::ORDER_VOLUME_COLUMN => $orderTotalVolume],
+            OrderConfig::ENTITY_ID_COLUMN . ' = ' . $orderID
+        );
+
+        $connection->update(
```

Comment:
You don't need to update grid table. It happens automatically. Please check \Magento\Sales\Model\ResourceModel\GridPool::refreshByOrderId